spec/models/solidus_stripe/create_intents_payment_service_spec.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe SolidusStripe::CreateIntentsPaymentService do
+  let(:service) { described_class.new(intent_id, stripe, controller) }
+
+  let(:stripe) {
+    Spree::PaymentMethod::StripeCreditCard.create!(
+      name: "Stripe",
+      preferred_secret_key: "sk_test_VCZnDv3GLU15TRvn8i2EsaAN",
+      preferred_publishable_key: "pk_test_Cuf0PNtiAkkMpTVC2gwYDMIg",
+      preferred_v3_elements: false,
+      preferred_v3_intents: true
+    )
+  }
+
+  let(:order) { create :order, state: :payment, total: 19.99 }
+
+  let(:intent_id) { "pi_123123ABC" }
+  let(:controller) { double(current_order: order.reload, params: params, request: spy) }
+
+  let(:params) do
+    {
+      spree_payment_method_id: stripe.id,
+      stripe_payment_intent_id: intent_id,
+      form_data: {
+        addressLine: ["31 Cotton Rd"],
+        city: "San Diego",
+        country: "US",
+        phone: "+188836412312",
+        postalCode: "12345",
+        recipient: "James Edwards",
+        region: "CA"
+      }
+    }
+  end
+
+  let(:intent) do
+    double(params: {
+      "id" => intent_id,
+      "charges" => {
+        "data" => [{
+          "payment_method_details" => {
+            "card" => {
+              "brand" => "visa",
+              "exp_month" => 1,
+              "exp_year" => 2022,
+              "last4" => "4242"
+            },
+          }
+        }]
+      }
+    })
+  end
+
+  describe '#call' do
+    subject { service.call }
+
+    before do
+      allow(stripe).to receive(:show_intent) { intent }
+      allow_any_instance_of(Spree::CreditCard).to receive(:require_card_numbers?) { false }
```

Comment:
Unfortunately, that would not be easy or simple (not even sure if it would be possible at all, but it would surely require much more tampering). Also, these objects do not exist yet at setup time as they are created later by another Solidus service object. For example, the instance of `Spree::CreditCard` spawns as a consequence of deeply nested parameter assignation  ( order -> payment -> source) and the instance method `#require_card_numbers?` is an AR callback on create. The second one is similar, as it spawns from an after save callback on the payment.

Reply:
Got it, no worries then!